behaviour/graql/language/insert.feature

```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
+      """
+    Given the integrity is validated
+    Given concept identifiers are
+      |      | check | value |
+      | HOM  | key   | ref:0 |
+      | BUR  | key   | ref:1 |
+      | SMI  | key   | ref:2 |
+      | NPP  | key   | ref:3 |
+      | eHOM | key   | ref:4 |
+      | eBUR | key   | ref:5 |
+      | eSMI | key   | ref:6 |
+    Given uniquely identify answer concepts
+      | x   | y   | z   | c   | xr   | yr   | zr   |
+      | HOM | BUR | SMI | NPP | eHOM | eBUR | eSMI |
+    Given get answers of graql query
+      """
+      match $r (employee: $x, employer: $c) isa employment; get;
+      """
+    Given uniquely identify answer concepts
+      | r    | x   | c   |
+      | eHOM | HOM | NPP |
+      | eBUR | BUR | NPP |
+      | eSMI | SMI | NPP |
+    When graql insert
+      """
+      match
+        $x isa employment;
+      insert
+        $x isa employment;
+      """
+    When the integrity is validated
+    Then get answers of graql query
+      """
+      match $r (employee: $x, employer: $c) isa employment; get;
+      """
+    Then uniquely identify answer concepts
+      | r    | x   | c   |
+      | eHOM | HOM | NPP |
+      | eBUR | BUR | NPP |
+      | eSMI | SMI | NPP |
+
+
+  Scenario: match-inserting only existing attributes is a no-op
+    Given get answers of graql query
+      """
+      insert
+      $x "Ash" isa name;
+      $y "Misty" isa name;
+      $z "Brock" isa name;
+      """
+    Given the integrity is validated
+    Given concept identifiers are
+      |     | check | value      |
+      | ASH | value | name:Ash   |
+      | MIS | value | name:Misty |
+      | BRO | value | name:Brock |
+    Given uniquely identify answer concepts
+      | x   | y   | z   |
+      | ASH | MIS | BRO |
+    Given get answers of graql query
+      """
+      match $x isa name; get;
+      """
+    Given uniquely identify answer concepts
+      | x   |
+      | ASH |
+      | MIS |
+      | BRO |
+    When graql insert
+      """
+      match
+        $x isa name;
+      insert
+        $x isa name;
+      """
+    When the integrity is validated
+    Then get answers of graql query
+      """
+      match $x isa name; get;
+      """
+    Then uniquely identify answer concepts
+      | x   |
+      | ASH |
+      | MIS |
+      | BRO |
+
+
+  ####################
+  # TRANSACTIONALITY #
+  ####################
+
+  Scenario: if any insert in a transaction fails with a syntax error, none of the inserts are performed
+    Given graql insert without commit
+      """
+      insert
+      $x isa person, has name "Derek", has ref 0;
+      """
+    Given graql insert throws
+      """
+      insert
+      $y qwertyuiop;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x isa person, has name "Derek"; get;
+      """
+    Then answer size is: 0
+
+
+  Scenario: if any insert in a transaction fails with a semantic error, none of the inserts are performed
```

Comment:
this makes me think we may want to decide to catch a "specific" type of exception, what do you think?

Reply:
Probably yes - but that could be applied to literally hundreds of our tests and I feel like it should be deferred for now.

Reply:
Raised https://github.com/graknlabs/verification/issues/60.